src/v0/index.js

```diff
@@ -152,8 +152,8 @@ export default () => {
         const slots = {};
         const recordSlots = node => (slots[node.getAttribute && node.getAttribute('slot') || '__default'] = true);
 
-        addedNodes.forEach(recordSlots);
-        removedNodes.forEach(recordSlots);
+        [...addedNodes].forEach(recordSlots);
```

Comment:
That said, [there's still some issues](https://toddmotto.com/ditch-the-array-foreach-call-nodelist-hack/#problem-4-creates-a-needless-array) with that approach. I think the archaic `if` method is the best choice considering performance is probably the most important thing here.